### ISSUE TITLE
Commenting out feature with its hotkey needed

### ISSUE THREAD

**Issue by jooyoungseo** [ajrgodfrey/WriteR#2]
Hello,

As you know, HTML comment markup is used for commenting out Rmd like below:

Having commenting out feature and its hotkey in WriteR would be instrumental for sure.

Thank you very much.

**Comment by ajrgodfrey** [comment]
Done.

Hot key is Alt+Shift+#
Please test and offer feedback.

**Comment by jooyoungseo** [comment]
That's great! Thanks for your hard work Dr. Godfrey.

Below are my testing results and follow-up suggestions:

 

1. The hotkey you assigned (Alt+Shift+3) is sort of hard to press. Would you mind considering replacing it with Alt+Semicolon? Alt+Semicolon is a hotkey that you use to comment out in Emacs, and it is really easy to press.

 

2. You assigned three dashes `<!--- comment --->` for both opening and closing tags. It does not matter though, but as far as I know, the standard is two dashes `<!-- comment -->`.

 

3. Last but not least, the current feature does not seem to support multi-line comments.

 

Let's say for example, you have three lines selected and then have pressed the hotkey for commenting out. The current feature only puts the opening and closing tag at the very beginning and the very end. However, Rmd does not support block comment; you should comment out each line. Thus, the ultimate feature should put a pair of opening and closing tags line by line, not as a whole.

 

Again, thank you very much for your kind work!

 

Sincerely,

 

Jooyoung

**Comment by ajrgodfrey** [comment]
Hi,

Hot key changed to Alt+q. I accept that the combination is perhaps difficult for some users. On my desktop keyboard it was easy enough, but on my own laptop it was more of a struggle. I chose q for quiet as more easily remembered than the suggested use of ;.

Third – removed in opening and closing tags. Agreed that they were not necessary.

Re multiple line comments. I think the only time I really do use this syntax is when I do want to comment out blocks. I’ve tested again, and the blocks of text were removed. My initial testing showed they were removed from the pdf as well which I found pleasantly surprising. No changes made, but please do test this thoroughly and report any problems.

Jonathan

**Comment by jooyoungseo** [comment]
One dash has been removed; however, the hotkey hasn’t changed on my end.

I have fetched and merged the latest commit.

 

Best,

JooYoung

**Comment by ajrgodfrey** [comment]
Sorted now

**Comment by jooyoungseo** [comment]
Two more suggestions:

 

1.	The current function puts extra return lines between before and after commented line. Would there be any way that we can just puts opening and closing tags for the desired line without putting any extra empty lines before and after? The reason is these extra lines are unnecessary and unintended by users.

 

2.	Would there be any way that we can make the function unqote the commented line? In other words, I am saying that I would like to have the function back and forth between commenting and uncommenting depending on whether the line is commented out or not.

**Comment by ajrgodfrey** [comment]
I played around and found the absence of blank lines more frustrating than their presence.
I managed to get the function to put the cursor at the end of the insertion string so that two presses of backspace will remove the line breaks and then up arrowing gets to the line before the tag inserted. I think deleting unnecessary space is easier than adding desired spacing.

The second request is mutually exclusive to the first so ought to be a new issue. Having said that, It will prove difficult to recognise if a line is within a chunk or between chunks. Adding a # sign between chunks will change text into a heading level 1 and ought to be avoided. For reasons of unforeseen consequences, I’m not in a hurry to add this feature.
